### PR TITLE
Merge -explain-types behavior into -explain behavior

### LINKED ISSUE #11637: HKT error report not that clear.
## Compiler version
3.0.0RC1
If you're not sure what version you're using, run `print scalaVersion` from sbt
(if you're running scalac manually, use `scalac -version` instead).

## Minimized example

```Scala
// This is OK
class FunctorImpl[Generic1[T] <: Iterable[T]]{}
class HKT3_1[Functor[Generic2[T]<:Set[T]]]{}
var h = new HKT3_1[FunctorImpl]();

// This is has error
class FunctorImpl[Generic1[T] <: Iterable[T]]{}
class HKT3_1[Functor[Generic2[T<:String]<:Set[T]]]{}
var h = new HKT3_1[FunctorImpl]();
// Error: Type argument FunctorImpl does not conform to upper bound [Generic2[T] <: Set[T]] =>> Any
```
## Output

```scala
No output
```

## Expectation
I am not sure, maybe `[Generic2[T<:String] <: Set[T]] =>> Any`? However, current error message is not enough.
And it would be nice to explain why it is not work in detailed.

### PR DESCRIPTION
**Pull request by odersky**
Eliminate -explain-types option. Error messages that can show
subtype traces now do so under the -explain mechanism.

Fixes #11637 

In fact, it turns out that we need to keep -explain-types since the sbt-tpolecat plugin, which is used in the community build, blows up without it. But it's now essentially an alias of -explain.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/core/TypeComparer.scala [hunk 1]
```diff
@@ -2595,8 +2591,8 @@ object TypeComparer {
       def addLow: Repr = approx | LoApprox
       def addHigh: Repr = approx | HiApprox
       def show: String =
-        val lo = if low then "LoApprox" else ""
-        val hi = if high then "HiApprox" else ""
+        val lo = if low then " (left is tightened)" else ""
+        val hi = if high then " (right is tigthened)" else ""
```
Comment (griggt) [inline compiler/src/dotty/tools/dotc/core/TypeComparer.scala:2595]:
```suggestion
        val hi = if high then " (right is tightened)" else ""
```

**Comment by griggt** [comment]
The `sbt-tpolecat` plugin (used by some community build projects) currently enables `-explain-types`:

https://github.com/DavidGregory084/sbt-tpolecat/blob/b28b564d9c5e9abfddda84be4ed4640d0ccbe383/src/main/scala/io/github/davidgregory084/TpolecatPlugin.scala#L51

File: compiler/src/dotty/tools/dotc/config/ScalaSettings.scala [hunk 2]
```diff
@@ -103,7 +103,7 @@ class ScalaSettings extends Settings.SettingGroup with CommonScalaSettings {
   val semanticdbTarget: Setting[String] = PathSetting("-semanticdb-target", "Specify an alternative output directory for SemanticDB files.", "")
 
   val deprecation: Setting[Boolean] = BooleanSetting("-deprecation", "Emit warning and location for usages of deprecated APIs.", aliases = List("--deprecation"))
-  val explain: Setting[Boolean] = BooleanSetting("-explain", "Explain errors in more detail.", aliases = List("--explain"))
+  val explain: Setting[Boolean] = BooleanSetting("-explain", "Explain errors in more detail.", aliases = List("--explain, -explain-types, --explain-types"))
```
Comment (griggt) [inline compiler/src/dotty/tools/dotc/config/ScalaSettings.scala]:
```suggestion
  val explain: Setting[Boolean] = BooleanSetting("-explain", "Explain errors in more detail.", aliases = List("--explain", "-explain-types", "--explain-types"))
```

Comment (odersky) [reply compiler/src/dotty/tools/dotc/config/ScalaSettings.scala]: Thanks!

**Comment by odersky** [comment]
@griggt Can I assign this to you for review? Thanks!

**Comment by SethTisue** [comment]
I'd previously suggested on https://github.com/lampepfl/dotty/pull/10737 that this really ought to be a `-V` option. Not sure if there is opposition to that, or if the question of naming just kind of got forgotten when that PR got closed.

**Comment by odersky** [comment]
@SethTisue So far, there exist no -V options in Scala 3. We should have a look at this later.

**Review by griggt — APPROVED** [review]
LGTM

**Pull request merged**